department-of-veterans-affairs/next-build
# Analytics update for next-build environments

# Description
What does this PR address?

Adds and updates the google analytics preview and auth for dev, staging and prod.
Replaces the <a> tag for the events Where link to a <va-link> component

## Ticket
Relates to #[656](https://github.com/department-of-veterans-affairs/next-build/issues/656). (or closes?)

## Developer Task

```[tasklist]
```

## Testing Steps
Explain the steps needed for testing

Navigate to an event page.
http://localhost:3000/outreach-and-events/events/68114/
or
{tugboat}/outreach-and-events/events/68114/

Go to the WHERE: link on the event page. Inspect that the <va-link> component is being used.

To test the analytics install [Adswerve](https://chromewebstore.google.com/detail/adswerve-datalayer-inspec/kmcbdogdandhihllalknlcjfpdjcleom?pli=1)
On chrome go to your settings in the dev inspector and turn on preserve logs under the network setting. Do not turn on any advance feature of Adswereve.

Check the data to confirm that GA4 events are firing

## QA steps
What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?

## Screenshots
Before:
<img width="448" alt="image" src="https://github.com/user-attachments/assets/69f2210e-564e-48b0-811e-ff9b21bf60d8">

After:

<img width="440" alt="image" src="https://github.com/user-attachments/assets/0257e8ab-c398-453a-b150-a9123f8cdd6a">


## Is this PR blocked by another PR?
- Add the `DO NOT MERGE` label
- Add links to additional PRs here:


# Reviewer

### Reviewing a PR

This section lists items that need to be checked or updated when making changes to this repository. 

## Standard Checks

```[tasklist]
```

## Merging an Approved Layout

When merging a layout, you must ensure that the content type has been turned on for `next-build` inside the `CMS`. This CMS flag must be turned on for editors to preview their work using the next build preview server.

Resource types (layouts) that have not been approved by design should NOT be pushed to production. Ensure that [slug.tsx](../src/pages/[[...slug]].tsx) does not include your resource type if it is not approved.

The status of layouts should be kept up to date inside [templates.md](./templates.md). This includes QA progress, development progress, etc. A link should be provided for where testing can occur.

## Merging a Non-Approved Layout

Your new resource type should not be included inside [slug.tsx](../src/pages/[[...slug]].tsx). Items added here will go into production once merged into the `main` branch. It is imperative that we do not push anything live that has not been approved.

Ensure that this layout has been added to the [templates.md](./templates.md) file with the current status of the work.
